cmd/controller-manager/main.go

```diff
@@ -109,15 +115,15 @@ func main() {
 
 	if utilfeature.DefaultFeatureGate.Enabled(features.SchedulerPreferences) {
 		for kind, schedulingType := range schedulingtypes.SchedulingTypes() {
-			err = schedulingpreference.StartSchedulingPreferenceController(kind, schedulingType.SchedulerFactory, config, fedNamespace, clusterNamespace, targetNamespace, stopChan, true)
+			err = schedulingpreference.StartSchedulingPreferenceController(kind, schedulingType.SchedulerFactory, config, fedNamespace, clusterNamespace, targetNamespace, stopChan, clusterAvailableDelay, clusterUnavailableDelay, false)
 			if err != nil {
 				glog.Fatalf("Error starting schedulingpreference controller for %q : %v", kind, err)
 			}
 		}
 	}
 
 	if utilfeature.DefaultFeatureGate.Enabled(features.CrossClusterServiceDiscovery) {
-		err = servicedns.StartController(config, fedNamespace, clusterNamespace, targetNamespace, stopChan, false)
+		err = servicedns.StartController(config, fedNamespace, clusterNamespace, targetNamespace, stopChan, clusterAvailableDelay, clusterUnavailableDelay, false)
```

Comment:
@shashidharatd please refer [this comment](https://github.com/kubernetes-sigs/federation-v2/issues/356#issuecomment-433157872), and confirm if its ok to have common delays for these three controllers. I have a feeling that dns based stuff might need larger timeouts. Although it might be ok to set them to common values, but probably the documentation of this flag provides a min value suitable to dns related code (for example, `setting the value lesser then xxxx might hinder with correct name resolution`)

Reply:
I think using common flags makes more sense to me. These flags mainly control when the reconciliation triggered and all the controllers are similar